CRM/Admin/Form/PaymentProcessor.php

```diff
@@ -400,4 +401,34 @@ public function updatePaymentProcessor(&$values, $domainID, $test) {
     civicrm_api3('PaymentProcessor', 'create', $params);
   }
 
+  /**
+   * Check if there is a record with the same name in the db.
+   *
+   * @param string $value
+   *   The value of the field we are checking.
+   * @param array $options
+   *   The daoName and fieldName (optional ).
+   *
+   * @return bool
+   *   true if object exists
+   */
+  public static function paymentProcessorNameExists($value, $options) {
+    $fieldName = 'name';
+    $daoName = CRM_Utils_Array::value(0, $options);
```

Comment:
I think the patch makes sense but stylistically 
1) it seems odd to pass in $daoName when there is no reason for it to be anything other than 'CRM_Financial_DAO_PaymentProcessor'
2) there is no reason to instantiate $config - that happens in lots of places randomly but for no reason that is still valid.

I think you could simplify the whole thing to

```
return civicrm_api3('PaymentProcessor', 'getcount' array(
  'id' => array('<>' => $id), 
  'domain_id' => CRM_Core_Config::domainID(), 
  'name' => $value,
));
```